job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -708,10 +727,15 @@ class WebApi(system: ActorSystem,
     import ContextSupervisor._
     val msg =
       if (context.isDefined) {
-        GetContext(context.get)
+        Left(GetContext(context.get))
       } else {
-        StartAdHocContext(classPath, contextConfig)
+        Right(StartAdHocContext(classPath, contextConfig))
       }
+    handleGetManager(msg)
+  }
+
+  private def handleGetManager(msgContainer: Either[GetContext, StartAdHocContext]): Option[ActorRef] = {
+    val msg = msgContainer.left.getOrElse(msgContainer.right.get)
```

Comment:
I'm not sure what an Either provides here, so this is an `Either[GetContext, StartAdHocContext]` ? Is this different than what you do above to switch the message based on a condition? you could just remove the `private def handleGetManager` entirely then.